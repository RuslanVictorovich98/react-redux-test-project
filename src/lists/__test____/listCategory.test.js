import React from 'react';
import {shallow} from 'enzyme';
import ConnectedListCategory, { ListCategory }  from '../listCategory';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

const initialState = {One: 'one'}
const mockStore = configureStore();
const mockFetchData = jest.fn();
const renderCategory = jest.fn();
let store, wrapper;

const props = {
    mainList: {products: [{bsr_category: 'Home & Kitchen'}], },
    category: mockFetchData,
    
}

beforeEach(()=>{
    store = mockStore(initialState);
    wrapper = shallow(<ListCategory store={store} {...props} renderCategory={renderCategory} />)
})

describe('TEST component ListCategory', () => {

    it('TEST spapshot to component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('ListCategory component', () => {
        expect(wrapper.exists()).toBe(true) // Returns whether or not any nodes exist in the wrapper.
    })

    it('TEST current loading first category', () => {
        const propsNew = {
            mainList: {products: ['Home & Kitchen']},
            category: mockFetchData,
            
        }
        wrapper = shallow(<ListCategory store={store} {...propsNew}/>)
        expect(wrapper.find('Link').first()).toHaveLength(1)
    })


    it('should call mockFetchData on category click and call `category` function', () => {
           
        wrapper.find('.data-value').simulate('click', {
                  currentTarget: { //1
                    innertText: ''
                }});
        expect(mockFetchData).toHaveBeenCalled();
        expect(wrapper.find('category')).toEqual({});
        
        wrapper.unmount();
      });
})

