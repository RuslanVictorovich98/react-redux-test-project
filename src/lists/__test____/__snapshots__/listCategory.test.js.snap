// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST component ListCategory TEST spapshot to component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <h4
    className="category-list"
  >
    <Link
      onClick={[Function]}
      to="/all-category"
    >
      All category
    </Link>
  </h4>
  <h4
    className="category-list data-value"
    key="0"
    onClick={[Function]}
  >
    <Link
      to="home-&-kitchen"
    >
      Home & Kitchen
    </Link>
  </h4>
</Router>
`;
